components/org.wso2.carbon.identity.application.authenticator.basicauth/src/main/java/org/wso2/carbon/identity/application/authenticator/basicauth/BasicAuthenticator.java

```diff
@@ -519,6 +520,16 @@ protected void processAuthenticationResponse(HttpServletRequest request,
             if (Boolean.parseBoolean(appendUserTenant)) {
                 loginIdentifierFromRequest = loginIdentifierFromRequest + "@" + context.getUserTenantDomain();
             }
+
+            /** FrameworkUtils.preprocessUsername will not append the tenant domain to username, if you are using
+             * email as username and EnableEmailUserName config is not enabled. So for a SaaS app, this config needs
+             * to be enabled to add the tenant domain of the application to email username if EnableEmailUserName
+             * is not enabled in the system.
+             **/
+            String appendTenant = runtimeParams.get(APPEND_TENANT_TO_USERNAME);
```

Comment:
can we rename to `appendAppTenantToUsername` to allign with other configs

Reply:
fixed